lib/sidekiq/throttled/strategy/base.rb

```diff
@@ -12,7 +12,14 @@ def limit(job_args = nil)
 
         def key(job_args)
           key = @base_key.dup
-          key << ":#{@key_suffix.call(*job_args)}" if @key_suffix
+          if @key_suffix
+            begin
+              key << ":#{@key_suffix.call(*job_args)}"
+            rescue Exception => e
+              Sidekiq.logger.error "Failed to get key suffix: #{e}"
+              raise e
+            end
+          end
           key
```

Comment:
We don't need to support Ruby < 2.6, thus we can (and I think) should simplify this:

``` ruby
def key(job_args)
  key = @base_key.dup
  return key unless @key_suffix

  key << ":#{@key_suffix.call(*job_args)}"
rescue => e
  Sidekiq.logger.error "Failed to get key suffix: #{e}"
  raise
end
```

I'm a bit against catching `Exception` here, though — we should catch StandardError only.

Reply:
Done. Thanks for the feedback!